src/plugins/janus_audiobridge.c

```diff
@@ -5559,6 +5583,131 @@ static json_t *janus_audiobridge_process_synchronous_request(janus_audiobridge_s
+		janus_audiobridge_room *audiobridge = g_hash_table_lookup(rooms,
+			string_ids ? (gpointer)room_id_str : (gpointer)&room_id);
+		if(audiobridge == NULL) {
+			janus_mutex_unlock(&rooms_mutex);
+			JANUS_LOG(LOG_ERR, "No such room (%s)\n", room_id_str);
+			error_code = JANUS_AUDIOBRIDGE_ERROR_NO_SUCH_ROOM;
+			g_snprintf(error_cause, 512, "No such room (%s)", room_id_str);
+			goto prepare_response;
+		}
+		/* A secret may be required for this action */
+		JANUS_CHECK_SECRET(audiobridge->room_secret, root, "secret", error_code, error_cause,
+			JANUS_AUDIOBRIDGE_ERROR_MISSING_ELEMENT, JANUS_AUDIOBRIDGE_ERROR_INVALID_ELEMENT, JANUS_AUDIOBRIDGE_ERROR_UNAUTHORIZED);
+		if(error_code != 0) {
+			janus_mutex_unlock(&rooms_mutex);
+			goto prepare_response;
+		}
+		janus_mutex_lock(&audiobridge->mutex);
+		if(audiobridge->destroyed) {
+			janus_mutex_unlock(&audiobridge->mutex);
+			janus_mutex_unlock(&rooms_mutex);
+			JANUS_LOG(LOG_ERR, "No such room (%s)\n", room_id_str);
+			error_code = JANUS_AUDIOBRIDGE_ERROR_NO_SUCH_ROOM;
+			g_snprintf(error_cause, 512, "No such room (%s)", room_id_str);
+			goto prepare_response;
+		}
+
+		/* Get list of started announcements and send a stop announcement notification*/
+		GList *items_to_remove = NULL;
+		GHashTableIter iterAnnc;
+		gpointer valueAnnc;
+		g_hash_table_iter_init(&iterAnnc, audiobridge->anncs);
+		while(g_hash_table_iter_next(&iterAnnc, NULL, &valueAnnc)) {
+			janus_audiobridge_participant *p = valueAnnc;
+			gboolean started = (p && p->annc && p->annc->started);
+			if(p)
+				janus_refcount_increase(&p->ref);
+			if(started) {
+				JANUS_LOG(LOG_INFO, "[%s] Announcement stopped (%s)\n", audiobridge->room_id_str, p->annc->id);
+				json_t *event = json_object();
+				json_object_set_new(event, "audiobridge", json_string("announcement-stopped"));
+				json_object_set_new(event, "room",
+				string_ids ? json_string(audiobridge->room_id_str) : json_integer(audiobridge->room_id));
+				json_object_set_new(event, "file_id", json_string(p->annc->id));
+				janus_audiobridge_notify_participants(p, event, TRUE);
+				json_decref(event);
+				/* Also notify event handlers */
+				if(notify_events && gateway->events_is_enabled()) {
+					json_t *info = json_object();
+					json_object_set_new(info, "event", json_string("announcement-stopped"));
+					json_object_set_new(info, "room",
+						string_ids ? json_string(audiobridge->room_id_str) : json_integer(audiobridge->room_id));
+					json_object_set_new(info, "file_id", json_string(p->annc->id));
+					gateway->notify_event(&janus_audiobridge_plugin, NULL, info);
+				}
+				items_to_remove = g_list_prepend(items_to_remove, p->annc->id);
+			}
+			if(p)
+				janus_refcount_decrease(&p->ref);
+		}
+
+		/* Get rid of the announcements */
+		json_t *listAnncRemoved = json_array();
+		for (GList *l = items_to_remove; l!= NULL; l = l->next) {
+			json_t *file_id = json_string(l->data);
+		    if( g_hash_table_remove(audiobridge->anncs, l->data))
+			{
```

Comment:
Code style: same for the `if`, there's an extra space that shouldn't be there before `g_hash_table_remove`.
The bracket of the check should be on the same line as the check, not below as you did here (`if(..) {`)